### ISSUE TITLE
qt5: cache qt&pyqt build results on Travis to speed up tests

### ISSUE THREAD

**Issue by kmike** [scrapinghub/splash#261]
See http://docs.travis-ci.com/user/caching/
//cc @sunu

**Comment by kmike** [comment]
hm, I guess the problem could be our use of sudo

**Comment by chekunkov** [comment]
<img width="1020" alt="job__961_1_-_scrapinghub_splash_-_travis_ci_and_ - _ _ printing _-_abbyy_lingvo-online" src="https://cloud.githubusercontent.com/assets/744331/9134384/3f327c8c-3d0c-11e5-9f0a-a4d93b1c5459.png">
 
@kmike we can speedup builds by [disabling sudo and using precompiled binaries](http://docs.travis-ci.com/user/migrating-from-legacy/?utm_source=legacy-notice&utm_medium=banner&utm_campaign=legacy-upgrade#Using-sudo-isn%E2%80%99t-possible-(right-now)), or we can wait, because header says "Using sudo isn’t possible (right now)" :)

**Comment by kmike** [comment]
@chekunkov as I recall, the problem is that Travis uses Ubuntu 12.04 which doesn't have official qt5 package, and to install qt5 from ppa in a new container-based Travis infrastructure this ppa must be whitelisted, but it is not. @sunu may know more details.

**Comment by sunu** [comment]
Yes basically we need the Qt5 ppa to be whitelisted first and then we can move to the container based new Travis infrastructure. I just submitted an issue https://github.com/travis-ci/apt-source-whitelist/issues/110. But I am not sure how and if at all they are handling the requests.
